src/ui/App.jsx

```
@@ -34,27 +43,63 @@ function App() {
+        if (detectionCache.current.has(trimmed)) {
+            cds = detectionCache.current.get(trimmed);
+        } else {
+            // Only run detection on new sentences or changed last sentence
+            // Note: For very long text, we might want to avoid re-checking the last sentence if it hasn't changed,
+            // but here we check everything not in cache.
+            // Since cache is persistent, this works for appending text.
+            cds = await detectCDs(trimmed, sensitivity);
+            detectionCache.current.set(trimmed, cds);
+        }
+
+        cds.forEach(cd => currentUniqueCDs.add(cd));
```

Biome (2.3.13)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)